templates/provider/kcm/templates/deployment.yaml

```diff
@@ -65,10 +68,12 @@ spec:
         - --zap-{{ $key }}={{ $value }}
         {{- end }}
         {{- end }}
-        {{- range $key, $value := .Values.controller.telemetry }}
-        {{- if not (eq (printf "%s" $value) "") }}
-        - --telemetry-{{ $key }}={{ $value }}
-        {{- end }}
+        - --telemetry-mode={{ $tel.mode }}
+        - --telemetry-concurrency={{ $tel.concurrency }}
+        - --telemetry-interval={{ $tel.interval }}
+        - --telemetry-jitter={{ $tel.jitter }}
+        {{- if $isLocalTelemetry }}
+        - --telemetry-baseDir={{ $baseDir }}
```

Comment:
Since we’ve moved away from the `--telemetry-{{ $key }}` approach anyway, it might be better to rename this parameter to `telemetry-base-dir` to avoid mixing camelCase with dash delimiters. But it's up to you.